web/src/compose_closed_ui.ts

```diff
@@ -36,11 +42,17 @@ export function get_recipient_label(message) {
                 // logic for what happens when you click the button,
                 // and not call into random modules with hacky fake
                 // "message" objects.
-                const user_ids = people.user_ids_string_to_ids_array(narrow_state.pm_ids_string());
+                const user_ids = people.user_ids_string_to_ids_array(narrow_state.pm_ids_string()!);
                 const user_ids_dicts = user_ids.map((user_id) => ({id: user_id}));
                 message = {
                     display_reply_to: message_store.get_pm_full_names({
                         type: "private",
+                        // This is expecting type DisplayRecipient with a
+                        // bunch of other things. The best way forward is
+                        // probably addressing the TODO above. A way to
+                        // replicate getting here is to start a DM message
+                        // to someone where there's no conversation yet,
+                        // and click to view the (empty) conversation.
```

Comment:
@andersk do you have ideas for how to deal with this?

Reply:
Can't you just have this function accept and work with an element of type  `Message | ComposeClosedMessage`? 

But the better answer is probably to just take the block at the bottom here:
```
    if (message) {
        if (message.stream_id && message.topic) {
            const stream = stream_data.get_sub_by_id(message.stream_id);
            if (stream) {
                return "#" + stream.name + " > " + message.topic;
            }
        } else if (message.display_reply_to) {
            return message.display_reply_to;
        }
    }
```

And make it a function, and each of these weird little code paths can either just do something simple like `return message.display_reply_to` or `return format_stream_recipient_label(stream_id, topic);`

So this particular block would just be 

```
return message_store.get_pm_full_names({
                        type: "private",
 ...
)
```

without any need to create a local variable called `message` and let it fall through.

Reply:
That cleans up the code a bit, but it doesn't address the issue of `message_store.get_pm_full_names` wanting a more complicated data structure than a list of user ids. I just pushed a commit doing a refactor to make this work, what do you think of that solution?

Reply:
Yeah that works, posted a couple small tweaks.